samcli/lib/build/app_builder.py

```diff
@@ -185,6 +185,10 @@ def _build_function(self, function_name, codeuri, runtime):
         # artifacts directory will be created by the builder
         artifacts_dir = str(pathlib.Path(self._build_dir, function_name))
 
+        if code_dir.lower().endswith(".zip"):
```

Comment:
Ahhh... I see. It's not building rather you want to skip functions that are zipped. 

This seems reasonable to me. The same comment I made on the layer_downloader file is also relevant here. We should skip any of the 'pre' built artifacts. I wonder if it is worth saying if `code_dir` is not a directory then skip? Thoughts?

Reply:
Yeah, skip zipping on 'build', and extract out for 'local invoke'. There's zip handling scattered around about three different places; I can take a shot at consolidating this if it'd be well received.